nabs/preprocessors.py

```diff
@@ -0,0 +1,94 @@
+import collections
+import logging
+import time
+
+from bluesky.plan_stubs import drop, save
+from bluesky.preprocessors import plan_mutator
+from bluesky.utils import make_decorator
+
+logger = logging.getLogger(__name__)
+
+
+class DropWrapper:
+    """
+    Replaces ``save`` messages with ``drop`` if the event is bad.
+
+    Parameters
+    ----------
+    filters: ``dict``, optional
+        A dictionary mapping from read key to function of one argument. This
```

Comment:
Either I'm confused or the docstring is wrong? Looks like you are expecting a list? I would recommend a `dict` so you can name your filters. Will make it more clear with multiple installed which is causing the drop.

Reply:
Oops, didn't update the docstring here. My mistake. I agree that naming the filters could be useful, I will make this change and synchronize the docstrings.